### PR TITLE
Implement LIKE pattern match lowering and evaluation

### PR DESCRIPTION
**Pull request by jpschorr**
Implement LIKE pattern match lowering and evaluation

Pattern compilation taken from https://github.com/partiql/partiql-lang-rust/pull/98

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-eval/src/pattern_match.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    #[test]
+    fn similar_match() {
+        let pat = similar_to_re_pattern("(b|c)%", Some('\\'));
+        let re = Regex::new(&pat).unwrap();
+        assert!(!re.is_match("abc"));
+
+        let pat = similar_to_re_pattern("%(b|d)%", Some('\\'));
+        let re = Regex::new(&pat).unwrap();
+        assert!(!re.is_match("abc"));
```
Comment (alancai98) [inline partiql-eval/src/pattern_match.rs:148]: Btw, this assertion fails in the GH Actions build: https://github.com/partiql/partiql-lang-rust/actions/runs/3859256107/jobs/6578608692#step:7:456

File: partiql-eval/src/pattern_match.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
+// TODO docs
+
+// TODO consider how to use the appropriate mechanisms to prevent exhaustion of
```
Comment (am357) [inline partiql-eval/src/pattern_match.rs:3]: You reckon as a short term measure we can put a hard limit for now?

Comment (jpschorr) [reply partiql-eval/src/pattern_match.rs:3]: added a limit of `const RE_SIZE_LIMIT: usize = 1 << 16;` for now. It's a bit arbitrarily chosen, as I don't have a good intuition for what limit corresponds to. 

Cf. https://docs.rs/regex/latest/regex/struct.RegexBuilder.html#method.size_limit

File: partiql-eval/src/eval.rs [hunk 3]
```diff
@@ -1028,6 +1029,39 @@ impl EvalExpr for EvalBetweenExpr {
...
+// Limit chosen somewhat arbitrarily, but to be smaller than the default of `10 * (1 << 20)`
+const RE_SIZE_LIMIT: usize = 1 << 16;
+
+impl EvalLikeMatch {
+    pub fn new(value: Box<dyn EvalExpr>, pattern: &str) -> Self {
+        let pattern = RegexBuilder::new(pattern)
+            .size_limit(RE_SIZE_LIMIT)
+            .build()
+            .expect("Like Pattern");
+        //let pattern = Regex::new(pattern).expect("Like Pattern");
```
Comment (alancai98) [inline partiql-eval/src/eval.rs:1048]: nit: commented out line

**Review by alancai98 — APPROVED** [review]

**Review by am357 — APPROVED** [review]

**Pull request merged**